### PR TITLE
[ci][bisect/2] bisect pipeline for macos test

### PR DESCRIPTION
**Pull request by can-anyscale**
Buildkite pipeline for macos test bisect. I'm reusing the existing 'bisect' pipeline

Test:
- CI
- https://buildkite.com/ray-project/bisect/builds/1030

### CODE REVIEW

File: .buildkite/bisect/config.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+buildkite_dirs:
+  - .buildkite/bisect
+env:
+  BUILDKITE_BAZEL_CACHE_URL: https://bazel-cache-dev.s3.us-west-2.amazonaws.com
+  RAYCI_SKIP_UPLOAD: "true"
+  # the following will be replaced by the actual values in the bisect pipeline
+  RAYCI_BISECT_TEST_NAME: __RAYCI_BISECT_TEST_NAME__
+  RAYCI_BISECT_PASSING_COMMIT: __RAYCI_BISECT_PASSING_COMMIT__
+  RAYCI_BISECT_FAILING_COMMIT: __RAYCI_BISECT_FAILING_COMMIT__
+  RAYCI_BISECT_TEST_TYPE: __RAYCI_BISECT_TEST_TYPE__
```
Comment (aslonnie) [inline .buildkite/bisect/config.yml:20]: > the following will be replaced by the actual values in the bisect pipeline

how? and why? why do you need to define them here?

File: .buildkite/bisect/config.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: ray-bisect
+artifacts_bucket: ray-ci-artifact-branch-public
+ci_temp: s3://ray-ci-artifact-branch-public/ci-temp/
+ci_work_repo: 029272617770.dkr.ecr.us-west-2.amazonaws.com/rayproject/citemp
+forge_prefix: cr.ray.io/rayproject/
```
Comment (aslonnie) [inline .buildkite/bisect/config.yml:5]: I feel that we are creating so many of these config files now..

File: .buildkite/bisect/config.yml (hunk 1 above)
Comment (can-anyscale) [reply .buildkite/bisect/config.yml:20]: yess, check out the bisect pipeline definition https://buildkite.com/ray-project/bisect/settings/steps

the `test_name`, etc. are provided as input through buildkite UI. I need a way to generate this config file based on that input in the init function. Not sure what is the best way to do so but this is what I can come up with. If there is a way to pass a list of environment variable that could work too.

File: ci/ray_ci/macos/macos_ci.sh [hunk 3]
```diff
@@ -104,17 +112,21 @@ _prelude() {
...
-  bazel run //ci/ray_ci/automation:test_db_bot -- core /tmp/bazel_event_logs
-  # Persist ray logs
-  mkdir -p /tmp/artifacts/.ray/
-  find /tmp/ray -path '*/logs/*' | tar -czf /tmp/artifacts/.ray/ray_logs.tgz -T -
-  # Cleanup runtime environment to save storage
-  rm -rf /tmp/ray
-  # Cleanup local caches - this should not clean up global disk cache
-  bazel clean
+  if [[ "${RAYCI_MACOS_SKIP_EPILOGUE-}" == 1 ]]; then
+    echo "SKIP_EPILOGUE is set, skipping epilogue"
```
Comment (can-anyscale) [inline ci/ray_ci/macos/macos_ci.sh:116]: setting this so we don't try to upload test data between bisect run

File: .buildkite/bisect/bisect.rayci.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+depends_on: 
+  - forge
+steps:
+  - name: macos test
+    commands:
+      - if [[ $(buildkite-agent meta-data get test-type) != "macos-test" ]]; then exit 0; fi
+      - MINIMAL_INSTALL=1 ./ci/ray_ci/macos/macos_ci.sh bisect 
+          $(buildkite-agent meta-data get test-name) 
+          $(buildkite-agent meta-data get passing-commit) 
+          $(buildkite-agent meta-data get failing-commit)
```
Comment (aslonnie) [inline .buildkite/bisect/bisect.rayci.yml:11]: double quote the `$( )` ?

File: .buildkite/bisect/bisect.rayci.yml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+group: bisect
+depends_on: 
+  - forge
+steps:
+  - name: macos test
+    commands:
+      - if [[ $(buildkite-agent meta-data get test-type) != "macos-test" ]]; then exit 0; fi
```
Comment (aslonnie) [inline .buildkite/bisect/bisect.rayci.yml:7]: double quote the `$( )`?

File: ci/ray_ci/macos/macos_ci.sh [hunk 6]
```diff
@@ -94,6 +94,14 @@ run_ray_cpp_and_java() {
   ./ci/ci.sh test_cpp || exit 42
 }
 
+bisect() {
+  # run bisect
+  test="$1"
+  passing_revision="$2"
+  failing_revision="$3"
```
Comment (aslonnie) [inline ci/ray_ci/macos/macos_ci.sh:101]: maybe just write `bazel run //ci/ray_ci/bisect:bisect_test -- "$@"`?

**Review by aslonnie — APPROVED** [review]

**Pull request merged**